src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -899,19 +899,65 @@ private bool CheckPropertyValueKind(SyntaxNode node, BoundExpression expr, BindV
             bool checkingReceiver,
             DiagnosticBag diagnostics)
         {
+            // check all arguments that are not passed by value
             if (!argRefKinds.IsDefault)
             {
-                for (var i = 0; i < args.Length; i++)
+                for (var argIdx = 0; argIdx < args.Length; argIdx++)
                 {
-                    if (argRefKinds[i] != RefKind.None && !CheckValueKind(args[i].Syntax, args[i], BindValueKind.ReturnableReference, false, diagnostics))
+                    if (argRefKinds[argIdx] != RefKind.None && !CheckValueKind(args[argIdx].Syntax, args[argIdx], BindValueKind.ReturnableReference, false, diagnostics))
                     {
                         var errorCode = checkingReceiver ? ErrorCode.ERR_RefReturnCall2 : ErrorCode.ERR_RefReturnCall;
-                        var parameterIndex = argToParamsOpt.IsDefault ? i : argToParamsOpt[i];
+                        var parameterIndex = argToParamsOpt.IsDefault ? argIdx : argToParamsOpt[argIdx];
                         var parameterName = parameters[parameterIndex].Name;
                         Error(diagnostics, errorCode, syntax, symbol, parameterName);
                         return false;
                     }
                 }
+
+                return true;
+            }
+
+            // check all "in" parameters 
+            for (var paramIdx = 0; paramIdx < parameters.Length; paramIdx++)
+            {
+                var parameter = parameters[paramIdx];
+
+                if (parameter.RefKind != RefKind.RefReadOnly)
+                {
+                    continue;
+                }
+
+                if (parameter.IsParams)
+                {
+                    break;
+                }
+
+                BoundExpression argument = null;
+                if (argToParamsOpt.IsDefault)
+                {
+                    if (paramIdx < args.Length)
+                    {
+                        argument = args[paramIdx];
+                    }
+                }
+                else
+                {
+                    for (int argIdx = 0; argIdx < args.Length; argIdx++)
+                    {
+                        if (argToParamsOpt[argIdx] == paramIdx)
+                        {
+                            argument = args[argIdx];
+                        }
+                    }
+                }
+
+                if (argument == null ||
+                    !CheckValueKind(argument.Syntax, argument, BindValueKind.ReturnableReference, false, diagnostics))
+                {
+                    var errorCode = checkingReceiver ? ErrorCode.ERR_RefReturnCall2 : ErrorCode.ERR_RefReturnCall;
+                    Error(diagnostics, errorCode, syntax, symbol, parameter.Name);
+                    return false;
```

Comment:
There is also the outer error which would say you cannot return the result of the call because one of the arguments is not returnable, then we report one of the arguments here, so that it is more clear where to start fixing. 
I think it is ok to return on the first error here. Even one error is optional. We could just return `false`.